test/Microsoft.Azure.WebJobs.Host.FunctionalTests/JobHostTests.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Azure.WebJobs.Host.FunctionalTests.TestDoubles;
+using Microsoft.Azure.WebJobs.Host.Storage;
+using Microsoft.Azure.WebJobs.Host.TestCommon;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Host.FunctionalTests
+{
+    public class JobHostTests
+    {
+        // Test that we can do static initialization before runtime initialization. 
+        [Fact]
+        public async Task StaticInit()
```

Comment:
I don't see why Script will need to execute any funky init calls on the config - that seems problematic to me. I think we can just rely on the fact that that init has been done internally at the right time by the underlying ScriptHost, and we flow the config around and use it naturally. I.e. I don't think we should have any new public extension methods on config.